modules/dlscript/godot.cpp

```diff
@@ -180,6 +181,16 @@ void GDAPI *godot_dlinstance_get_userdata(godot_object *p_instance) {
 	return NULL;
 }
 
+void GDAPI godot_get_global_constants(godot_dictionary *p_constants) {
```

Comment:
`_clear()` isn't to free. `_destroy()` is to free. Also it's not really more explicit.

```cpp
godot_dictionary d = godot_get_global_constants();

// this is the same
godot_dictionary d;
godot_get_global_constants(&d);
```

You can just assign to a more complex and managed datastructure just fine.
In fact, most C compilers implement struct return types that way anway. (reserving stack space before the call (or any other memory location for that matter), writing into the reserved memory, then return) So it's semantically almost identical to your pointer argument implementation, this is just nicer to read